### ISSUE TITLE
Compatibility with WCAG-EM

### ISSUE THREAD

**Issue by AlainVagner** [accessibility-luxembourg/ReferentielAccessibiliteWeb#20]
In the frame of the [presentation of this framework at NL Design Systems Week 2025](https://www.youtube.com/watch?v=GCaCmbHDQOE), one question has been asked about the compability of RAWeb with WCAG-EM. WCAG-EM is important in the Netherlands, because it is required for all audits done for the public sector AFAIK.

I have found some divergences between the RAWeb (and RGAA) method and WCAG-EM 1. 

# Proposal 

So far, it does not seem mandatory to support WCAG-EM in Luxembourg, there is no mention of it in our Laws, in the WAD and EAA, and in their implementing acts (ex: 2018/1524).
If we would like to fully support WCAG-EM in RAWeb, we could integrate for example the following changes:

- Sections 3 and 4.c: These sections describe how to construct the sample with a structured component and a random component. While these two components are mentioned for defining a sample, some elements of the methodology are missing. The testing methodology page could potentially be supplemented with these elements. For example, a summary of the requirements of WCAG-EM Part 3 could be included, along with a reference to it.
- 5.a: We could add the following to our templates:
    - the sponsor's name
    - further detail the site's scope
    - or explain in the methodology that, for us, the scope is always all the pages of a site available at a given URL? Do we sometimes encounter other cases? Does the scope correspond to the Context field in our system?
- Identify the pages of the same process within the sample.

What is your opinion on the compability with WCAG-EM? What do you think of the proposed changes?

# Detailed analysis

## 1.a define the scope of the website: Define the target [website](https://www.w3.org/TR/WCAG-EM/#website) according to [Scope of Applicability](https://www.w3.org/TR/WCAG-EM/#applicability), so that for each [web page](https://www.w3.org/TR/WCAG-EM/#webpage) it is unambiguous whether it is within the scope of evaluation or not.

I haven't found anything in RAWeb and RGAA on this topic. In the templates that we propose, we have some fields for the name of the company and the URL of the website. An extra field named "context" is available. Is this enough?

## 1.b Define the Conformance Target:  Select a target WCAG 2.0 [conformance level](https://www.w3.org/TR/WCAG20/#cc1) ("A", "AA", or "AAA") for the evaluation.
For us, the target is defined by default in EN 301 549. We give the information about which version of the Framework do we use, ex: RAWeb 1.
Cf Word template.

## 1.c: Define an Accessibility Support Baseline : Define the web browser, assistive technologies and other [user agents](https://www.w3.org/TR/WCAG20/#useragentdef) for which features provided on the website are to be [accessibility supported](https://www.w3.org/TR/WCAG20/#accessibility-supporteddef).
For us, the baseline is provided by default and can be redefined. We have one part on this point in the Word template.

## 2: Explore the Target Website
There are here 5 requirements which structure this part.
This is implicitely done but not documented in the RAWeb method.
For the step 2.d, we have une section in the report which mentions all the technologies available on the website, this one is also available in the WCAG-EM tool.

## 3.a Include a Structured Sample : Select web pages and web page states that reflect all identified (1) [common web pages](https://www.w3.org/TR/WCAG-EM/#common), (2) [essential functionality](https://www.w3.org/TR/WCAG-EM/#functionality), (3) types of web pages, (4) web technologies relied upon, and (5) other relevant web pages.
RAWeb proposes a list of predefined pages, including full processes, or which particular technologies.

## 3.b: Include a Randomly Selected Sample : Select a random sample of web pages and web page states, and include them for auditing.
This is the case, we have a sample of 10% of random pages

## 3.c : Include Complete Processes : Include all web pages and web page states that are part of a [complete process](https://www.w3.org/TR/WCAG-EM/#complete) in the selected sample.
This is the case

## 4.a: Check All Initial Web Pages: Check that each web page and web page state in the selected sample that is not within or the end of a complete process conforms to each of the five WCAG 2.0 conformance requirements at the target conformance level.
Here we have one difference:

> Note: WCAG 2.0 Success Criteria are each formulated as a "testable statement that will be either true or false when applied to specific web content". When there is no content presented to the user that relates to specific Success Criteria (for example, no video on the web page), then the Success Criteria are "[satisfied](https://www.w3.org/TR/WCAG/#satisfiesdef)" according to WCAG 2.0. Optionally, an evaluation report can specifically indicate Success Criteria for which there is no relevant content, for example, with "not present". [Understanding Conformance](https://www.w3.org/TR/UNDERSTANDING-WCAG20/conformance.html) provides more background and guidance.

if one criterion is not applicable, then it is compliant for WCAG-EM, but the information that the type of content was not present can be optionally provided. The WCAG-EM tool has two states more than we have: "cannot tell" and "not present". I suppose it is then no problem to have one "not applicable" state which includes "not present". We could possibly include these two extra states.

The concept of an alternative version is taken into account.  The integration of "accessibility support" and the non-interference principle is there (for RAWeb at least). 
Regarding conformance, it is not possible to be 100% compliant if third-party content are not compliant. The concept of partially conformance seems to be different from the one of the WAD. As we legally have to implement the WAD, the priority shall be given on the WAD here.
See the [definition of partial conformance in WCAG](https://www.w3.org/TR/UNDERSTANDING-WCAG20/conformance.html#uc-partial-third-head)

## Step 4.b: Check All Complete Processes: Check that all interaction for each web page and web page state along a [complete process](https://www.w3.org/TR/WCAG20/#cc3) conforms to each of the five WCAG 2.0 conformance requirements at the target conformance level.
It is implemented in RAWeb

## 4.c: Compare Structured and Random Samples: Check that each web page and each web page state in the randomly selected sample does not show types of content and outcomes that are not represented in the structured sample.
This not foreseen yet in RAWeb. The goal is to see if the sample is representative enough, by detecting if we find new accessibility issues in the random pages which were not previously detected in the other part of the samples. This is an iterative process.

## 5.a: Document the Outcomes of Each Step: Document each outcome of the steps defined in [Step 1: Define the Evaluation Scope](https://www.w3.org/TR/WCAG-EM/#step1), [Step 2: Explore the Target Website](https://www.w3.org/TR/WCAG-EM/#step2), [Step 3: Select a Representative Sample](https://www.w3.org/TR/WCAG-EM/#step3), and [Step 4: Audit the Selected Sample](https://www.w3.org/TR/WCAG-EM/#step4).
These are the mandatory informations:
 
- About the Evaluation 
    - Name of the [evaluator](https://www.w3.org/TR/WCAG-EM/#evaluator) : we have it in the Word template, object of the audit
    - Name of the [evaluation commissioner](https://www.w3.org/TR/WCAG-EM/#commissioner) : we don't have it (but is it available in the WCAG-EM tool?)
    - Date for the evaluation (completion date or duration period) : we have this
- Evaluation Scope 
    - Scope of the website defined in [Step 1.a: Define the Scope of the Website](https://www.w3.org/TR/WCAG-EM/#step1a) We have only the URL of the site, here the name of the site and the scope are required. Example of scope: all the pages available at this URL http://example.org
    - Conformance target defined in [Step 1.b. Define the Conformance Target](https://www.w3.org/TR/WCAG-EM/#step1b) We have this in the Excel template
    - Accessibility support baseline defined in [Step 1.c: Define an Accessibility Support Baseline](https://www.w3.org/TR/WCAG-EM/#step1c) we have this in the Word template
- Website Exploration 
    - Web technologies relied upon identified in [Step 2.d: Identify Web Technologies Relied Upon](https://www.w3.org/TR/WCAG-EM/#step2d) We have this in the Word template
- Representative Sample, we don't have it
    - Web pages selected through structured sampling in [Step 3.a: Include a Structured Sample](https://www.w3.org/TR/WCAG-EM/#step3a) 
    - Randomly selected web pages and selection method used in [Step 3.b: Include a Randomly Selected Sample](https://www.w3.org/TR/WCAG-EM/#step3b) 
    - Complete processes selected in [Step 3.c: Include Complete Processes](https://www.w3.org/TR/WCAG-EM/#step3c) 
- Sample Audited, we have it but not with all the details
    - Evaluation outcomes from [Step 4.a: Check All Initial Web Pages](https://www.w3.org/TR/WCAG-EM/#step4a)
    - Evaluation outcomes from [Step 4.b: Check All Complete Processes](https://www.w3.org/TR/WCAG-EM/#step4b)
    - Evaluation outcomes from [Step 4.c: Compare Structured and Random Samples](https://www.w3.org/TR/WCAG-EM/#step4c)
 

> Note: Depending on the desired granularity of the report documentation, the outcomes of [Step 4: Audit the Selected Sample](https://www.w3.org/TR/WCAG-EM/#step4) may be provided for each evaluated web page and web page state, or aggregated over the entire sample. Reports should include at least one example for each conformance requirement and WCAG 2.0 Success Criterion not met. It is also good practice for evaluators to indicate issues that occur repeatedly.

We have this in RAWeb.

## 5.c: Provide an Evaluation Statement (Optional)
We have this in RAWeb 
We have it, although the template requested by the WAD is slightly different. There's no mention of the baseline, for example, nor of the technologies used.

## 5.d: Provide an Aggregated Score (Optional)
We have this in RAWeb

## 5.e: Provide Machine-Readable Reports (Optional)
This is planned with our JSON format. For now, it's just not available in the EARL format recommended by the W3C.
We also have metadata in the accessibility statements generated with our tool, which allows for the extraction of structured information via a script.
